var/spack/repos/builtin/packages/libpeas/package.py

```diff
@@ -0,0 +1,97 @@
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('gettext', type='build')
+    depends_on('pkg-config', type='build')
+    depends_on('atk')
+    depends_on('intltool@0.40.0:')
+    depends_on('xmlto')
+    depends_on('perl')
+    depends_on('perl-xml-parser')
+    depends_on('glib@2.10:')
+    depends_on('gobject-introspection')
+    depends_on('libffi')
+    depends_on('gtkplus')
+    depends_on('gdk-pixbuf')
+    depends_on('pango')
+    depends_on('gnome-common')
+    depends_on('py-pygobject@3:')
+    depends_on('python@3:')
+
+    def setup_environment(self, spack_env, run_env):
+        # Let
+        #
+        # python = self.spec['python']
+        # prefix = python.prefix
+        # pyversion = python.version.up_to(2)
+        # python_lib_path = os.path.join(prefix, 'Frameworks',
+        #                                'Python.framework', 'Versions',
+        #                                pyversion)
+        #
+        # self.spec['python'].libs.ld_flags returns (on macOS)
+        # '-L{0} -lPython'.format(python_lib_path)
+        #
+        # e.g., for python@3.7.4 on macOS via Homebrew, python_lib_path is
+        # /usr/local/opt/python/Frameworks/Python.framework/Versions/3.7
+        #
+        # This directory is correct for many purposes, but libpeas uses the
+        # link flag '-lpython{0}m'.format(pyversion) and does not use an
+        # appropriate -L flag to locate this library, so the correct -L flag
+        # must be appended to LDFLAGS. Furthermore, this library is not found
+        # in python_lib_path.  However, pkg-config returns the correct
+        # directory, so pkg-config is used to generate the correct paths for
+        # LDFLAGS.
+        pkg_config = which('pkg-config')
+        python_prefix = self.spec['python'].prefix.lib.pkgconfig
+        python_pc_file = os.path.join(python_prefix, 'python3.pc')
+        python_ldflags = pkg_config('--libs', python_pc_file, output=str)
+
+        spack_env.append_path('LDFLAGS',
+                              python_ldflags)
+
+    def autoreconf(self, spec, prefix):
+        autoreconf_args = ['-ivf']
+        aclocal_pkg_list = ['pkg-config',
+                            'gettext',
+                            'intltool',
+                            'glib',
+                            'gobject-introspection',
+                            'gnome-common',
+                            'gtkplus']
+        aclocal_path = os.path.join('share', 'aclocal')
+
+        for pkg in aclocal_pkg_list:
+            autoreconf_args += ['-I',
+                                os.path.join(spec[pkg].prefix, aclocal_path)]
+
+        autoreconf = which('autoreconf')
+        autoreconf(*autoreconf_args)
+
+    def configure_args(self):
+        args = ['--disable-silent-rules',
+                '--enable-gtk',
+                '--enable-python3',
+                '--disable-python2']
```

Comment:
I wonder if we should support both Python 2 and 3? Personally, I only use 3, and 2 will be obsolete by the end of the year, so I don't really care either way.

Reply:
@adamjstewart I based the flags off of Homebrew's `libpeas` formula, and omitted Python 2 support due to the Python 2 end-of-life date you mentioned.